src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+            }
+        } else {
+            false
+        }
+    }
+
+    // If enabled, returns the path of the metrics_file, otherwise returns None.
+    fn get_metrics_file_path(&mut self) -> Result<Option<&str>, GooseError> {
+        // If metrics are disabled, or running in Manager mode, there is no
+        // metrics file, exit immediately.
+        if self.configuration.no_metrics || self.attack_mode == GooseMode::Manager {
+            return Ok(None);
+        }
+
+        // If --metrics-file is set, return it.
+        if !self.configuration.metrics_file.is_empty() {
+            return Ok(Some(&self.configuration.metrics_file));
+        }
+
+        // If GooseDefault::MetricFile is set, return it.
+        if let Some(default_metrics_file) = &self.defaults.metrics_file {
+            return Ok(Some(default_metrics_file));
+        }
+
+        // Otherwise there is no metrics file.
+        Ok(None)
+    }
+
+    // Configure metrics log format.
+    fn set_metrics_format(&mut self) -> Result<(), GooseError> {
+        if self.configuration.metrics_format.is_empty() {
+            if let Some(metrics_format) = &self.defaults.metrics_format {
+                self.configuration.metrics_format = metrics_format.to_string();
+            } else {
+                self.configuration.metrics_format = "json".to_string();
+            }
+        } else {
+            // Log format isn't relevant if metrics aren't enabled.
+            if self.configuration.no_metrics {
+                return Err(GooseError::InvalidOption {
+                    option: "--no-metrics".to_string(),
+                    value: "true".to_string(),
+                    detail: "The --no-metrics flag can not be set together with the --metrics-format option.".to_string(),
+                });
+            }
+            // Log format isn't relevant if log not enabled.
+            else if self.get_metrics_file_path()?.is_none() {
+                return Err(GooseError::InvalidOption {
+                    option: "--metrics-format".to_string(),
+                    value: self.configuration.metrics_format.clone(),
+                    detail: "The --metrics-file option must be set together with the --metrics-format option.".to_string(),
+                });
+            }
+        }
+
+        let options = vec!["json", "csv", "raw"];
+        if !options.contains(&self.configuration.metrics_format.as_str()) {
+            return Err(GooseError::InvalidOption {
+                option: "--metrics-format".to_string(),
+                value: self.configuration.metrics_format.clone(),
+                detail: format!(
+                    "The --metrics-format option must be set to one of: {}.",
+                    options.join(", ")
+                ),
+            });
+        }
+
+        Ok(())
+    }
+
+    // If enabled, returns the path of the metrics_file, otherwise returns None.
+    fn get_debug_file_path(&self) -> Result<Option<&str>, GooseError> {
```

Comment:
@LionsAd Should we use `get_` or `calculate_`, or overload `configuration` for consistency?